pkg/vault/contracts/Vault.sol

```diff
@@ -486,22 +476,14 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
         );
 
         // 7) Off-chain events.
-        // Since the swapFeeAmountScaled18 (derived from scaling up either the amountGiven or amountCalculated)
-        // also contains the rate, undo it when converting to raw.
-        locals.swapFeeAmountRaw = locals.swapFeeAmountScaled18.toRawUndoRateRoundDown(
-            poolData.decimalScalingFactors[locals.swapFeeIndex],
-            poolData.tokenRates[locals.swapFeeIndex]
-        );
-
         emit Swap(
             vaultSwapParams.pool,
             vaultSwapParams.tokenIn,
             vaultSwapParams.tokenOut,
             amountInRaw,
             amountOutRaw,
             swapState.swapFeePercentage,
-            locals.swapFeeAmountRaw,
-            locals.swapFeeToken
+            locals.totalSwapFeeAmountRaw
```

Comment:
Consider adding a comment defining `totalSwapFeeAmountRaw` as the total swap fee, which later gets divided between the protocol, pool creator, and LPs. It's potentially a bit confusing, as we usually report the aggregate fees (e.g., in the getters), and this is a change from what we did before.

Reply:
that's already in the struct definition :)